src/vm/ilmarshalers.h

```diff
@@ -1962,6 +1962,18 @@ class ILWSTRMarshaler : public ILMarshaler
     }
 #endif // _DEBUG
 
+    
+    virtual bool SupportsArgumentMarshal(DWORD dwMarshalFlags, UINT* pErrorResID)
+    {
+        if (IsOut(dwMarshalFlags) && !IsByref(dwMarshalFlags))
```

Comment:
What about if marked by `out`? Does the `IsByRef()` return true for both the `ref` and `out` keywords? Based on this usage I am also assuming that `IsOut()` is for the attribute _not_ the keyword. Is that correct?

Reply:
At a metadata level there's no difference between a `[Out] ref` and `out`. `IsByRef` returns true for both `ref` and `out` keywords. So, this check is only true for an `[Out]` by value, not `[Out] ref` or `out`.